data/helpers.d/setting

```diff
@@ -179,27 +189,72 @@ ynh_webpath_register () {
 # Requires YunoHost version 3.7.0 or higher.
 ynh_permission_create() {
     # Declare an array to define the options of this helper.
-    local legacy_args=pua
-    local -A args_array=( [p]=permission= [u]=url= [a]=allowed= )
+    local legacy_args=puAhaltP
+    declare -A args_array=( [p]=permission= [u]=url= [A]=additional_urls= [h]=auth_header= [a]=allowed= [l]=label= [t]=show_tile= [P]=protected= )
     local permission
     local url
+    local additional_urls
+    local auth_header
     local allowed
+    local label
+    local show_tile
+    local protected
     ynh_handle_getopts_args "$@"
     url=${url:-}
+    additional_urls=${additional_urls:-}
+    auth_header=${auth_header:-}
     allowed=${allowed:-}
+    label=${label:-}
+    show_tile=${show_tile:-}
+    protected=${protected:-}
 
     if [[ -n $url ]]
     then
-        url="'$url'"
-    else
-        url="None"
+        url=",url='$url'"
+    fi
+
+    if [[ -n $additional_urls ]]
+    then
+        additional_urls=",additional_urls=['${additional_urls//';'/"','"}']"
+    fi
+
+    if [[ -n $auth_header ]]
+    then
+        if [ $auth_header == "true" ]
+        then
+            auth_header=",auth_header=True"
+        else
+            auth_header=",auth_header=False"
+        fi
     fi
```

Comment:
Considering the final value is a boolean.
What's the behavior if auth_header is empty ?

Same for other variable used the same way

Reply:
We use the default value define [here](https://github.com/YunoHost/yunohost/blob/permission_protection/src/yunohost/permission.py#L260-L263)

Reply:
Then comment and explain.
When using the helper, before using true or false, the packager should know what are the default values.

Reply:
The default value are documented here https://github.com/YunoHost/yunohost/pull/861/files#diff-bb250d2955b70f7920149b8c2942ec59R169

Reply:
What about `show_tile` ?
Add a comment around those lines, the behavior should be clear.